code/src/components/GameStart.js

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { game } from '../reducers/game';
+import { startFetch } from '../reducers/reusable';
+import { GameHeader } from '../lib/GameContainerStyle';
+import { GameButton } from '../lib/Button';
+import { InputForm } from '../lib/GameStartStyle';
+
+export const GameStart = () => {
+    const [userNameInput, setUserNameInput] = useState("");
+    const dispatch = useDispatch();
+    const gameDetails = useSelector(store => store.game.gameDetails)
+    
+    const handleSubmit = e => {
+        e.preventDefault();
+        dispatch(game.actions.updateUserName(userNameInput))
+        dispatch(startFetch(userNameInput))
+    }
+
+
+    return (
+        <>
+        {!gameDetails.coordinates && ( 
+            <section>
+        <GameHeader>Want to play?</GameHeader>
+        <InputForm>
+            <label
+                value='username'
+            />
+            <input
+                type='text'
+                placeholder='Type username'
+                onChange={e => setUserNameInput(e.target.value)}
+            />
+            <GameButton onClick={handleSubmit} width={100} disabled={!userNameInput}>Let's go!</GameButton>
```

Comment:
Really good that you disabled the button if the user hasn't typed in a user name (will change that in our own application as well! 👍)